src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/DeprecationGroup.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bridge\PhpUnit\DeprecationErrorHandler;
+
+final class DeprecationGroup
+{
+    /**
+     * @var int
+     */
+    private $count = 0;
+
+    /**
+     * @var array<string,DeprecationNotice> keys are messages
+     */
+    private $deprecationNotices = [];
+
+    /**
+     * @param string $message
+     * @param string $class
+     * @param string $method
+     */
+    public function addNoticeFromObject($message, $class, $method)
+    {
+        if (!isset($this->deprecationNotices[$message])) {
```

Comment:
This looks redundant

Reply:
It isn't, otherwise line 39 would not be worth much. The goal here is to know how many times a given notice occurs.

Reply:
I don't follow, exactly same if is in `deprecationNotice` method

Reply:
Oh you're right, I didn't notice that 👍